src/networking.c

```diff
@@ -2395,10 +2408,13 @@ int processMultibulkBuffer(client *c) {
 
         c->multibulklen = ll;
 
-        /* Setup argv array on client structure */
-        if (c->argv) zfree(c->argv);
-        c->argv_len = min(c->multibulklen, 1024);
-        c->argv = zmalloc(sizeof(robj*)*c->argv_len);
+        /* Setup argv array on client structure.
+         * Create new argv if space is insufficient or if we need to allocate it gradually. */
+        if (unlikely(c->multibulklen > c->argv_len || c->multibulklen > 1024)) {
+            zfree(c->argv);
+            c->argv_len = min(c->multibulklen, 1024);
```

Comment:
if c->multibulken > 1024, the c->argv_len is 1024? There is not enough space to store argv.

here is c->argv_len = c->multibulklen?

Reply:
no, c->argv will be reallocated when the space is not enough in https://github.com/redis/redis/blob/64a40b20d906f019a32fcb227c938768079dd9ed/src/networking.c#L2508-L2512